test/test_constraint.jl

```diff
@@ -1887,4 +1887,58 @@ function test_set_normalized_coefficient_quadratic_batch()
     return
 end
 
+function test_symmetric_matrix_inequality()
+    model = Model()
+    @variable(model, x[1:2, 1:2], Symmetric)
+    @variable(model, y[1:2, 1:2], Symmetric)
+    set_start_value.(x, [1 2; 2 3])
+    set_start_value.(y, [6 4; 4 7])
+    g = [x[1, 1] - y[1, 1], x[1, 2] - y[1, 2], x[2, 2] - y[2, 2]]
+    for set in (Nonnegatives(), Nonpositives(), Zeros())
+        c = @constraint(model, x >= y, set)
```

Comment:
Add a test for `x - y in set`

Reply:
Ah. @mlubin, I remember now. 

We can't support `x - y in Nonnegatives()` because then we can't tell it apart from `@constraint(model, x >= y)` which was the whole point in the first place.

Reply:
> We can't support `x - y in Nonnegatives()`

What do you mean ? You mean `x >= y, Nonnegatives()` ?

Reply:
No I mean `@constraint(model, x - y in Nonnegatives())` because this hits the same codepath as `@constraint(model, x >= y)`.

Reply:
Unless I intercept the `GreaterThan(false)` method, but then I think the last time I looked that makes ambiguities.

Reply:
Ah, that's the part I didn't get. It does make sense to intersect the `GreaterThan` then because having `AbstractArray`-in-`GreaterThan` is ambiguous and shouldn't be allowed while `AbstractArray`-in-`Nonnegatives` should be allowed.
So neither `@constraint(model, x - y in GreaterThan(false))` nor `@constraint(model, x >= y)` will work.

Reply:
Ah. I forgot. We switched from `GreaterThan(false)` _to_ `Nonnegatives`. So we can't even intercept. `x >= y` is exactly equal to `x - y in Nonnegatives`

Reply:
Ouch, that was probably not ideal. Using `Nonnegatives` by default kind of means we interpret `>=` as elementwise by default.

Reply:
Maybe we could add a layer of indirection to fix that without being breaking